docs/user/agent/add-node/node-joiner.sh

```diff
@@ -0,0 +1,144 @@
+nodeJoinerPullspec=$(oc adm release info --image-for=baremetal-installer --registry-config="$pullSecretFile")
+
+# Use the same random temp file suffix for the namespace.
+namespace=$(echo "openshift-node-joiner-${pullSecretFile#/tmp/nodejoiner-}" | tr '[:upper:]' '[:lower:]')
+
+# Create the namespace to run the node-joiner, along with the required roles and bindings.
+staticResources=$(cat <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ${namespace}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: node-joiner
+  namespace: ${namespace}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: node-joiner
+rules:
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - clusterversions
+  - proxies
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - configmaps
+  - nodes
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: node-joiner
+subjects:
+- kind: ServiceAccount
+  name: node-joiner
+  namespace: ${namespace}
+roleRef:
+  kind: ClusterRole
+  name: node-joiner
+  apiGroup: rbac.authorization.k8s.io
+EOF
+)
+echo "$staticResources" | oc apply -f -
+
+# Generate a configMap to store the user configuration
+oc create configmap nodes-config --from-file=nodes-config.yaml="${nodesConfigFile}" -n "${namespace}" -o yaml --dry-run=client | oc apply -f -
+
+# Run the node-joiner pod to generate the ISO
+nodeJoinerPod=$(cat <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: node-joiner
+  namespace: ${namespace}
+  annotations:
+    openshift.io/scc: anyuid
+  labels:
+    app: node-joiner    
+spec:
+  restartPolicy: Never
+  serviceAccountName: node-joiner
+  securityContext:
+    seccompProfile:
+      type: RuntimeDefault
+  containers:
+  - name: node-joiner
+    imagePullPolicy: IfNotPresent
+    image: $nodeJoinerPullspec
+    volumeMounts:
+    - name: nodes-config
+      mountPath: /config
+    - name: assets
+      mountPath: /assets
+    command: ["/bin/sh", "-c", "cp /config/nodes-config.yaml /assets; HOME=/assets node-joiner add-nodes --dir=/assets --log-level=debug; sleep 600"]    
+  volumes:
+  - name: nodes-config
+    configMap: 
+      name: nodes-config
+      namespace: ${namespace}
+  - name: assets
+    emptyDir: 
+      sizeLimit: "4Gi"
+EOF
+)
+echo "$nodeJoinerPod" | oc apply -f -
+
+while true; do 
+  if oc exec node-joiner -n "${namespace}" -- test -e /assets/exit_code >/dev/null 2>&1; then
+    break
+  else 
+    echo "Waiting for node-joiner pod to complete..."
+    sleep 10s
+  fi
+done
+
+res=$(oc exec node-joiner -n "${namespace}" -- cat /assets/exit_code)
+if [ "$res" = 0 ]; then
+  echo "node-joiner successfully completed, extracting ISO image..."
+  oc cp -n "${namespace}" node-joiner:/assets/node.x86_64.iso node.x86_64.iso
+else
+  oc logs node-joiner -n "${namespace}"
+  echo "node-joiner failed"
+fi
+
+echo "Cleaning up"
```

Comment:
Not required because would be eventually cleaned up because it is in /tmp, but would be nice to also clean up the pull secret file at the end.

Reply:
Not sure to follow, the `cleanup` function will be invoked when the scripts ends (for any reason, see [here](https://github.com/openshift/installer/pull/8242/files#diff-2c6496e2d523a6836ab7af9d4667347b4476f2c6cdc9f1e8d5c29822f70ea91eR20))

Reply:
Sorry, I missed that.